common/src/async_interface.rs

```diff
@@ -0,0 +1,62 @@
+use crate::value::Value;
+use tokio::sync::mpsc::{self, UnboundedReceiver, UnboundedSender};
+
+/// Inter-modules communication
+pub struct AsyncInterface<Req, Res> {
+    tx_request: Value<UnboundedSender<Req>>,
```

Comment:
Nit: If you're always returning pairs, perhaps have two `Option<>`-s of a two-element tuple each, not four one-element `Option<>`-s?

(I am using `Option<>` here since we know what the `Value<>` really is.)